### ISSUE TITLE
Kafka Binding v1.8: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)

### ISSUE THREAD

**Issue by Andemki** [dapr/dapr#5342]
## What version of Dapr?

> 1.8.x

## Expected Behavior

Dapr and Azure Event Hub integration using Kafka binding work without errors

## Actual Behavior

I've been using bindings.kafka component for recieving messages from Azure Event Hub in my application. Everything work properly in Dapr versions 1.6.x and 1.7.x. But when I updated dapr components to version 1.8, the sidecar container (daprd) of my application's pod failed to start and I received the following error message: 

> level=fatal msg="process component my-eventhub-receiver error: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)". 

When I downgraded Dapr to version 1.6, my application which uses bindings.kafka component started to work properly again.  So the question is how to make it work in v.1.8.
Yaml output of my bindings.kafka component:
````
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  annotations:
    meta.helm.sh/release-name: my-eventhublistener
    meta.helm.sh/release-namespace: mynamespace
  creationTimestamp: "2022-09-07T17:08:46Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  name: my-eventhub-receiver
  namespace: mynamespace
  resourceVersion: "108720906"
  uid: 203b893d-7a9f-47d9-a97c-b3f99a2fdd01
spec:
  metadata:
  - name: brokers
    value: my-eh-namespace.servicebus.windows.net:9093
  - name: topics
    value: my-eh-data
  - name: consumerGroup
    value: my-eventhublistener
  - name: authRequired
    value: "true"
  - name: saslUsername
    value: $ConnectionString
  - name: saslPassword
    secretKeyRef:
      key: connectionString
      name: my-eventhublistener-eventhub
  type: bindings.kafka
  version: v1
````

## Steps to Reproduce the Problem
1) Create a kafka.binding component to receive messages from Event Hub in your application
2) Deploy your application and the component to the Kubernetes
3) Check the daprd component is not in Running state.

**Comment by skyao** [comment]
This bug should be very similar to this bug, which is happens in kafka pubsub:

https://github.com/dapr/components-contrib/issues/2182

In dapr v1.8, I have done code refactory for kafka binding and kafka puhsub to remove the duplicated code. See :

- https://github.com/dapr/components-contrib/pull/1695: Refactory kafka pubsub components for code reuse
- https://github.com/dapr/components-contrib/pull/1696: Refactory kafka binding to reuse the kafka common code extracting from kafka pubsub component

During this code refactory, I have make a mistaken about setting config.Net.TLS.Enable=true when tls ca is not set, see the detail in https://github.com/dapr/components-contrib/issues/2182.

This bug should be fixed after rollback the code to v1.7.2.

But there is still a small different between kafka binding and kafka pubsub about TLS setting:

This is the code of  kafka binding  in v1.7.2 to set SASL:

```go
func updateAuthInfo(config *sarama.Config, saslUsername, saslPassword string) {
	config.Net.SASL.Enable = true
	config.Net.SASL.User = saslUsername
	config.Net.SASL.Password = saslPassword
	config.Net.SASL.Mechanism = sarama.SASLTypePlaintext

	config.Net.TLS.Enable = true
	// nolint: gosec
	config.Net.TLS.Config = &tls.Config{
		// InsecureSkipVerify: true,
		ClientAuth: 0,
	}
}
```

After code refactory (since v1.8), kafka binding will reuse the code in kafka pubsub, the code will be :

```go
func updatePasswordAuthInfo(config *sarama.Config, saslUsername, saslPassword string) {
	config.Net.SASL.Enable = true
	config.Net.SASL.User = saslUsername
	config.Net.SASL.Password = saslPassword
	config.Net.SASL.Mechanism = sarama.SASLTypePlaintext
}

func updateTLSConfig(config *sarama.Config, metadata *kafkaMetadata) error {
	if metadata.TLSDisable || metadata.AuthType == noAuthType {
		config.Net.TLS.Enable = false
		return nil
	}
	config.Net.TLS.Enable = true  // this line will go back to set TLS.Enable = true even TLSCaCert is empty (as your case)

	if !metadata.TLSSkipVerify && metadata.TLSCaCert == "" {
                // !!! the difference is here, in kafka pubsub it will return if  TLSCaCert is not set (as your case)
		return nil
	}

        // nolint: gosec
	config.Net.TLS.Config = &tls.Config{InsecureSkipVerify: metadata.TLSSkipVerify, MinVersion: tls.VersionTLS12}
	if metadata.TLSCaCert != "" {
		caCertPool := x509.NewCertPool()
		if ok := caCertPool.AppendCertsFromPEM([]byte(metadata.TLSCaCert)); !ok {
			return errors.New("kafka error: unable to load ca certificate")
		}
		config.Net.TLS.Config.RootCAs = caCertPool
	}

	return nil
```

The difference is that in kafka binding v1.7.2: 

```go
	config.Net.TLS.Config = &tls.Config{
		// InsecureSkipVerify: true,
		ClientAuth: 0,
	}
```

In kafka pubsub of v1.7.2 and both in kafka pubsub and binding of v1.8 : 

```go
config.Net.TLS.Config = nil 
```

**Comment by mukundansundar** [comment]
> Hello @skyao. Thank you for the explanation.
> 
> > This bug should be fixed after rollback the code to v1.7.2.
> 
> 
> 
> We would like to use v.1.8 because we need Injector watchdog functionality which is only available starting from 1.8 version.
> 
> Could you please tell me when this issue will be fixed in v.1.8?
> 
> 

@Andemki the fix for this issue has been released in https://github.com/dapr/dapr/releases/tag/v1.8.5

Please try it out and let us know if it works for you.

**Comment by Andemki** [comment]
@mukundansundar @skyao Hello. Yes, I confirm that this issue has been resolved in Dapr v.1.8.5. Kafka binding for Azure EventHub Listener works properly for us in this version. Thank you very much!
